engine/engine.go

```diff
@@ -557,3 +559,18 @@ func matchesAllSelectors(m RuleMeta, selectors ...RuleSelector) bool {
 	}
 	return true
 }
+
+func deduplicateLabels(labels *[]string) {
```

Comment:
@AdiAkhileshSingh15 Thank you for taking this! can we not use pointers here, and just return a new array?

Reply:
> @AdiAkhileshSingh15 Thank you for taking this! can we not use pointers here, and just return a new array?

We surely can. Initially, I leaned towards pointers for efficiency, but as returning a new array of unique labels can make the code more readable, we can surely do it that way.